packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart

```diff
@@ -246,6 +275,12 @@ abstract class GoogleMapsFlutterPlatform extends PlatformInterface {
     throw UnimplementedError('takeSnapshot() has not been implemented.');
   }
 
+  /// Set the [MapGetTileCallback] for the map, which will be called
+  /// when a [Tile] is requested for an added [TileProvider].
+  void setGetTileCallback({@required int mapId, MapGetTileCallback callback}) {
```

Comment:
I'm not sure I understand why we need this method?
When we pass the platform overlays to the platform side in `updateTileOverlays` we already pass each overlay with it's associated TileProvider, is it possible for the platform side to use that information to cache the provider for each overlay?

Adding this extra method for setting a global provider for all overlays adds an opportunity for multiple (and potentially inconsistent) source of truth regarding the tile for a given overlay at a given coordinate.

Or am I completely missing something?